artifact/keyfactor/signer_test.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2023 The Keyfactor Command Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package keyfactor
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"testing"
+)
+
+func TestNewSignServerSigner(t *testing.T) {
+	// Get the signer name from the environment
+	workerName := os.Getenv("SIGNSERVER_WORKER_NAME")
+	if workerName == "" {
+		t.Fatal("SIGNSERVER_WORKER_NAME must be set")
+	}
+
+	// Use NewSignServerSigner to create a new SignServerSigner
+	signer, err := NewSignServerSigner(workerName)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Check that the signer is not nil
+	if signer == nil {
+		t.Fatal("signer is nil")
+	}
+
+	// Check that the signer has a properly configured client
+	assert.NotNil(t, signer.client)
+
+	// Check that the signer has a properly configured worker name
+	assert.Equal(t, workerName, signer.workerName)
+
+	t.Run("sign", func(t *testing.T) {
```

Comment:
If you look at the current code, you see that we flavor having the tests as top level `func TestSomething` and then, if any, have the share code in some helper function.

It would be nice to align, but I won't push much for this.

Reply:
I've split each test case into its own function per the patterns used by the rest of the project. 
<img width="524" alt="image" src="https://github.com/mendersoftware/mender-artifact/assets/49427552/c3ef082f-00cc-4031-831d-cb4d78ee996f">

Reply:
Thank you